libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -1736,6 +1736,11 @@ impl NodeAPI for Greenlight {
         }
         Ok((hints, has_public_channel))
     }
+
+    async fn has_active_channel_to(&self, lsp_info: &LspInformation) -> NodeResult<bool> {
```

Comment:
This will create several requests for each lsp. It is better IMO to call node api once and get the list of channels which then can be compared to the lsp list.